rp2040-hal/src/clocks/macros.rs

```diff
@@ -0,0 +1,210 @@
+#[macro_use]
+
+macro_rules! int_division {
+    ($name:ident, $div:ident, $u:ty) => {
+        impl IntegerDivision for $name {
+            fn set_int_div(&mut self, div: usize) {
+                unsafe { self.shared_dev.get() }.$div.write(|w| unsafe {
+                    w.int().bits(div as $u);
+                    w
+                });
+            }
+        }
+    };
+}
+
+macro_rules! frac_division {
+    ($name:ident, $div:ident, $u:ty) => {
+        impl FractionDivision for $name {
+            fn set_frac_div(&mut self, div: usize) {
```

Comment:
Using `write` instead of `modify` will clobber the rest of this register. This is an issue throughout but is important here as we can't set both the integer and fractional part of the divider.

Reply:
Yeah, when I wrote the code I was still confused about the difference between `wirte` and `modify`. Those need to be changed.